src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -19,35 +20,84 @@
+    finalBallSensor = new DigitalInput(Constants.finalBallSensorChannel); 
 
-    intakeMotorControllerTwo.getSparkMax().follow(intakeMotorControllerTwo.getSparkMax());
+    DigitalInput[] sensorArray = {initialBallSensor, middleBallSensor, finalBallSensor};
+
+    // Remove invert=true parameter if wheels aren't running correctly
+    //CDSWheelControllerOne.getSparkMax().follow(intakeMotorControllerOne.getSparkMax());
+    //CDSWheelControllerTwo.getSparkMax().follow(intakeMotorControllerOne.getSparkMax(), true);
+    intakeMotorControllerTwo.getSparkMax().follow(intakeMotorControllerOne.getSparkMax());
   }
 
-  public void IntakeSwitch(boolean on){    
-    if (on){
-      double intakeSmartSpeed = SmartDashboard.getNumber("Belt Speed", Constants.CDSBeltSpeed);
-      
-      intakeMotorControllerOne.getSparkMax().set(intakeSmartSpeed);
+  public void toggleIntake(boolean reverse) {
+    if (reverse) {
+      intakeMotorControllerOne.getSparkMax().set(-Constants.intakeMotorSpeed);
+      SmartDashboard.putString("Intake Motor Direction", "Reverse");
+      SmartDashboard.putNumber("Intake Motor Speed", -Constants.intakeMotorSpeed);
+    } else {
+      intakeMotorControllerOne.getSparkMax().set(Constants.intakeMotorSpeed);
+      SmartDashboard.putString("Intake Motor Direction", "Forward");
       SmartDashboard.putNumber("Intake Motor Speed", Constants.intakeMotorSpeed);
+    }
+  }
+
+  public void stopIntake() {
+    intakeMotorControllerOne.getSparkMax().set(0.0);
+    SmartDashboard.putNumber("Intake Motor Speed", 0.0);
+  }
+
+  public boolean getDirection() {
+    // true = inverted, false = forward
+    if (intakeMotorControllerOne.getSparkMax().get() > 0) {
+      return false;
     } else {
-      intakeMotorControllerOne.getSparkMax().set(0);
-      SmartDashboard.putNumber("Intake Motor Speed", 0);
+    return true;
     }
   }
-  
-  public void ForwardIntake(){
-    intakeMotorControllerOne.getSparkMax().setInverted(false);
-    SmartDashboard.putString("Intake Motor Direction", "Forward");
+
+  public boolean[] getBeamBreakStatus() {
+    boolean initialStatus = initialBallSensor.get();
+    boolean middleStatus = middleBallSensor.get();
+    boolean finalStatus = finalBallSensor.get();
+    boolean[] beamBreakArray = {initialStatus, middleStatus, finalStatus};
+    return beamBreakArray;
   }
 
-  public void ReverseIntake(){
-    intakeMotorControllerOne.getSparkMax().setInverted(true);
-    SmartDashboard.putString("Intake Motor Direction", "Reverse");
+  public int getBallCount() {
+    boolean[] beamBreakStatuses = this.getBeamBreakStatus();
+    if (!beamBreakStatuses[0]) {
+      if (this.getDirection()){
+        ballCount--;
+      } else {
+        ballCount++;
+      }
+    }
+    return ballCount;
   }
 
-  
+  public void expelBalls() {
+    if (ballCount > 2) {
+      this.toggleIntake(true);
+      while (this.getBallCount() > 2) {
+      }
+      this.toggleIntake(false);
```

Comment:
End state should probably be to stop the motor rather than switch its direction. Consider renaming toggleIntake to something like runIntake. 

Since it seems like the intent here is to potentially override a button press, so maybe using the while loop is a good way to do that. But currently this implements a blocking wait which may cause issues with other subsystems if it takes too long. I'm not sure how well the scheduler handles that.

Reply:
The while loop could possibly trigger motor watch dogs which will kill all motor outputs. Looping like this should be avoided.